manual/manual/cmds/intf-c.etex

```diff
@@ -189,19 +189,15 @@ These files reside in the "caml/" subdirectory of the OCaml
 standard library directory, which is returned by the command
 "ocamlc -where" (usually "/usr/local/lib/ocaml" or "/usr/lib/ocaml").
 
-By default, header files in the "caml/" subdirectory give only access
-to the public interface of the OCaml runtime. It is possible to define
-the macro "CAML_INTERNALS" to get access to a lower-level interface,
-but this lower-level interface is more likely to change and break
-programs that use it.
-
-{\bf Note:} It is recommended to define the macro "CAML_NAME_SPACE"
-before including these header files. If you do not define it, the
-header files will also define short names (without the "caml_" prefix)
-for most functions, which usually produce clashes with names defined
-by other C libraries that you might use. Including the header files
-without "CAML_NAME_SPACE" is only supported for backward
-compatibility.
+{\bf Note:} It is highly recommended to define the macro "CAML_NAME_SPACE"
+on the command line with "-DCAML_NAME_SPACE" when including any of these
+header files. Otherwise, it is your responsability to ensure that
+the macro "CAML_NAME_SPACE" is defined bedore any of these header files.
+If you do not define it, the header files will also define short names
+(without the "caml_" prefix) for most functions, which usually produce
+clashes with names defined by other C libraries that you might use.
+Including the header files without "CAML_NAME_SPACE" is only supported
+for backward compatibility and may be removed in the future.
```

Comment:
I'm sorry I didn't follow the long discussions about this. What does this mean in practice ?

1. I should not do `ocamlc mystub.c` but `ocamlc -ccopt  '-DCAML_NAME_SPACE' mystub.c` ? Seems a bit annoying.
2. I can still `ocamlc mystub.c` but I should not complain if I use an unprefixed name and it no longer works in the future. 
3. ???

Reply:
So the paragraph is unclear, but the forward-compatible options are more:

1) use ocamlc -ccopt '-DCAML_NAME_SPACE' (I agree that this is a bit annoying)
2) define `CAML_nAME_SPACE` before importing any `caml/` headers

And the option that is only supported for backward compatibility and might disappear at some point:

3) use short names

Reply:
Note that in my 2. the idea was that at certain point `ocamlc file.c` would add the `-DCAML_NAME_SPACE` itself when short names are no longer supported.

I'm wondering if the instructions should then rather not be (and the sample code in the chapter be updated accordingly): before `#include`ing any `<caml/.*h>` you have to issue a `#define CAML_NAME_SPACE`. That way we can simply have people `ocamlc mystub.c` their c files.

The note should then simply mention that it is possible to do it without the #define but that this may bring clashes and short names and will not be supported in the future.

Reply:
I tend to prefer @dbuenzli suggestion. Suggesting macro definition is better than the explicit arguments since that lives with the code and hence, is explicit / tracked in source control, can be caught in code review, etc. With the compiler flag, I can imagine users may find it difficult to figure out how to get their high-level build tools (dune, ocamlbuild, etc) to pass the option down to the compiler. 

s/rather not be/rather be?

Reply:
@dbuenzli , thanks for reminding me of updating the code examples!

Throwing an error might be too much, but we could start by printing a notification. I agree that it will probably better to introduce `CAML_NAME_SPACE` earlier and only mention the command line definition path as an alternative path.